src/com/less04/homework/MyLinkedList.java

```
@@ -0,0 +1,118 @@
+        if (size == 0) {
+            first = newNode;
+        }
+        last.nextElement = newNode;
+        last = newNode;
+        size++;
+    }
+
+    @Override
+    public void remove(int i) {
+        try {
+            Node<T> x = getNode(i);
+            if (x != null) {
+                remove(x);
+            }
+        } catch (Exception e) {
+            System.out.println("Can't remove element with this index");
+        }
+    }
+
+    private void remove(Node<T> node) {
+        Node<T> prev = node.prevElement;
+        Node<T> next = node.nextElement;
+        if (prev.currentElement == null) {
+            first = node.nextElement;
+        } else if (next == null) {
+            last = node.prevElement;
+        } else {
+            prev.nextElement = next;
+            next.prevElement = prev;
+            node.nextElement = null;
+            node.prevElement = null;
+        }
+        node.currentElement = null;
+        size--;
+
+    }
+
+    private Node<T> getNode(int index) throws Exception {
+        if (index < 0 || index >= size) {
+            throw new Exception("Index is out of bound!");
+        } else if (index < size / 2) {
+            Node<T> item = first;
+            for (int i = 0; i < index; i++) {
+                item = item.nextElement;
+            }
+            return item;
+        } else {
+            Node<T> item = last;
+            for (int i = size - 1; i > index; i--) {
+                item = item.prevElement;
+            }
+            return item;
+        }
+    }
+
+    @Override
+    public void clear() {
+        for (int i = 0; i < size; i++) {
+            remove(0);
+        }
+    }
+
+    @Override
+    public int size() {
+        return size;
+    }
+
+    @Override
+    public T get(int i) {
+        try {
+            return getNode(i).currentElement;
+        } catch (Exception e) {
+            System.out.println("Can't remove element with this index");
```

It is a bad idea. It is better to throw an exception if  i<0 or i>size-1